### PR TITLE
DTSPO-8223  - improve readme and fix workflow

### PR DESCRIPTION
**Pull request by msl8r**
### Change description ###
Mainly updates to do with the feedback in the comments in DTSPO-8223, but the deploy workflow was also failing more then not, so I made some updates to that to get it working reliably again. 

Changes:
- Added a watch flag to message generator script
- Fixed the create queue step in the deploy workflow by dropping `NewDefaultAzureCredential` and explicitly specifying the ways of getting a token. 
- Updated dependencies 
- Updated docs with more clear instructions on how to successfully load message and how to find the resources created.
- Tweaked the format of credentials that the azure login action uses as `--sdk-auth` is deprecated and will be removed. Mentioned here: https://github.com/marketplace/actions/azure-login#configure-deployment-credentials

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,18 +51,80 @@ To reduce the amount of code and the complexity around making sure that both the
...
+These resources tell the cluster how to authenticate with the chosen Azure Service Bus and when to scale the recipe-receiver pods so that they can clear the queue quicker.
+
+Resources for the recipe-receiver app that are deployed to the SDS cluster will be deployed into `toffee` namespace whereas resources deployed to the CFT clusters will go to the `cnp` namespace.
+
+### Viewing the created resources 
+
+Before starting, you will need to set your kubectl context to the correct AKS cluster.
+
+#### Get scaledJob and triggerAuthentication resources
+The `-l` flags is being used to get the exact resources for chosen PR, you should up the PR number or if you prefer you can omit the `-l` flag and its argument to get all the Keda resources in that namespace.
```
Comment (timja) [inline README.md:66]: what do you mean by 'up the PR number'

File: README.md [hunk 2]
```diff
@@ -51,18 +51,80 @@ To reduce the amount of code and the complexity around making sure that both the
...
+CFT - list the pods for PR 34:
+
+`kubectl get pods -l app.kubernetes.io/name=recipe-receiver-pr-34-function -n cnp`
+
+Again, you can get all the pods by omitting the `-l` flag.
+
 ## Loading a queue with messages
 
-The script to load messages into a queue is located in the messageGenerator directory. You can either run the binary if you're on macOS or use `go run main.go`. For the script to work you'll need to have the `Azure Service Bus Data Sender` role assigned to your Azure account.
+The script to load messages into a queue is located in the messageGenerator directory, you'll need go installed to run it. If go isn't already installed on your machine you can follow the [installation instructions](https://go.dev/doc/install) to get the latest version.
```
Comment (timja) [inline README.md:97]:
```suggestion
The script to load messages into a queue is located in the messageGenerator directory, you'll need `go` installed to run it. If go isn't already installed on your machine you can follow the [installation instructions](https://go.dev/doc/install) to get the latest version.
```

File: README.md (hunk 1 above)
Comment (msl8r) [reply README.md:66]: update 😂 🤦🏼

Comment (msl8r) [reply README.md:66]:
```suggestion
The `-l` flags is being used to get the exact resources for chosen PR, you should update the PR number or if you prefer you can omit the `-l` flag and its argument to get all the Keda resources in that namespace. 
```

**Review by timja — APPROVED** [review]

**Pull request merged**